src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -36,6 +36,19 @@
   private static final Schema DEFAULT_INFER_SCHEMA_TYPE = Schema.OPTIONAL_STRING_SCHEMA;
   public static final String SCHEMA_NAME_TEMPLATE = "inferred_name_%s";
 
+  public static Schema inferDocumentSchema(final BsonDocument document) {
+    SchemaBuilder builder = SchemaBuilder.struct();
```

Comment:
Can we DRY this up with the code in inferSchema?

Reply:
I had hoped to but  it ended up making the code more complex and harder to read. I'll take another pass.

Reply:
Well  that was easy now the document schema inference was already in its own method.